RecoHI/HiJetAlgos/plugins/HiBadParticleFilter.cc

```diff
@@ -0,0 +1,285 @@
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+//
+// class declaration
+//
+
+class HiBadParticleFilter : public edm::stream::EDProducer<> {
```

Comment:
I dunno, the rest of the EDProducers in the directory are stream, not global.   Is one better than the other?

Reply:
stream modules are run once per stream; global modules are run only once per job
In this case it can be a global producer (analogously to when you made it a global filter in the previous version)

Reply:
Ok, but I can't judge which is better in this case.  Are you advising me to make it global?

Reply:
`run` here probably means `created`.
every kind of module still runs at most once per event.